openedx/frontend-app-course-authoring
# feat: pagination studio home for courses

### Description

This PR adds pagination for the studio home view and makes minor changes to each course card.


**NOTE:** This needs to be activated by the environment variable `ENABLE_HOME_PAGE_COURSE_API_V2` otherwise, it will continue using the old course list

enable this feature flag
`new_studio_mfe.use_new_home_page`

### DEMO



https://github.com/openedx/frontend-app-course-authoring/assets/134975835/cf4fe6b9-495b-4fe7-9e97-61a02f356d90



<img width="1459" alt="Screen Shot 2024-03-22 at 1 08 51 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/134975835/2afb27ee-9d88-43e1-9e8c-82968e82c878">



### Course cards



BEFORE        |  NOW
:-------------------------:|:-------------------------:
![](https://github.com/openedx/frontend-app-course-authoring/assets/134975835/bbc78f47-b162-4c74-bd8c-190dcdf91eeb)  |  ![](https://github.com/openedx/frontend-app-course-authoring/assets/134975835/2afb27ee-9d88-43e1-9e8c-82968e82c878)




### Supporting information
To review a little bit more about this implementation, please check this:
https://openedx.atlassian.net/wiki/spaces/OEPM/pages/4032856101/Studio+Home+incremental+upgrades+-+product+approach


### How to test it
1. Set up your edx-platform
    ```
     cd path-to-your-edx-platform
     git fetch origin pull/34175/head:MJG/course-home-api-v2-pagination
     git checkout MJG/course-home-api-v2-pagination
    ```
2. Check it out this branch inside frontend-app-course-authoring
    ```
     cd path-to-frontend-app-course-authoring
     git fetch origin pull/825/head:jv/feat-home-studio-pagination
     git checkout jv/feat-home-studio-pagination
     npm start
    ```

3. The default items per page is 10 so you would need to create more than 10 courses

#### You can create this tutor plugin to enable it
```yml
name: mfe_course_authoring_config
version: 0.1.0
patches:
  mfe-lms-development-settings: |
    MFE_CONFIG["ENABLE_HOME_PAGE_COURSE_API_V2"]=True

  mfe-lms-production-settings: |
    MFE_CONFIG["ENABLE_HOME_PAGE_COURSE_API_V2"]=True

```
